azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/Subnet.java

```diff
@@ -71,6 +93,7 @@
     interface Definition<ParentT> extends
         DefinitionStages.Blank<ParentT>,
         DefinitionStages.WithAddressPrefix<ParentT>,
+        DefinitionStages.WithNetworkSecurityGroup<ParentT>,
         DefinitionStages.WithAttach<ParentT> {
     }
 
```

Comment:
`WithAttach<ParentT>` extends from `DefinitionStages.WithNetworkSecurityGroup<ParentT>`. In this case we don't have to extends `Definition<ParentT>` from `DefinitionStages.WithNetworkSecurityGroup<ParentT>` since  `Definition<ParentT>` already extends `WithAttach<ParentT>`, right?

One standard i am trying to follow is:

`Definition<ParentT>` extends all interfaces in the REQUIRED chain then followed by `DefinitionStages.WithAttach<ParentT> `.  `DefinitionStages.WithAttach<ParentT> ` extends all interfaces in OPTIONAL chain, this way we don't duplicate interfaces `extends`.